crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -2564,6 +2678,9 @@ impl<'db> TypeInferenceBuilder<'db> {
                 report_invalid_exception_cause(&self.context, cause, cause_type);
             }
         }
+
+        let range = cause.as_ref().map_or(raise.range(), |cause| cause.range());
+        self.record_return_type(Type::Never, range);
```

Comment:
I don't think we need this? By definition, `Never` is assignable to all types, so it is impossible for a `Never` return to ever emit a diagnostic, no matter what the annotated return type is.

Removing this change doesn't cause any tests to fail; unless we can write a counter-example test, we should remove this.

Reply:
Pushed this change.